Justfile

```diff
@@ -0,0 +1,50 @@
+# Justfile
+# 'just' is a command runner, like a modern 'make'
+# Install instructions at https://github.com/casey/just?tab=readme-ov-file#installation
+# Usage is as simple as 'just test' to run all the tests we care about.
+
+# The default recipe that runs if you type 'just' with no arguments
+# List available commands
+default:
+    @just --list --unsorted
+
+# For all the recipes in this file, don't echo the commands to the terminal
+set quiet
+
+# Run the most recently built slosh executable
+slosh:
+    @if [ -f ./target/debug/slosh ] && [ -f ./target/release/slosh ]; then \
+        if [ ./target/debug/slosh -nt ./target/release/slosh ]; then \
+            ./target/debug/slosh; \
+        else \
+            ./target/release/slosh; \
+        fi; \
+    elif [ -f ./target/debug/slosh ]; then \
+        ./target/debug/slosh; \
+    elif [ -f ./target/release/slosh ]; then \
+        ./target/release/slosh; \
+    else \
+        cargo build && ./target/debug/slosh; \
+    fi
+
+# Run all the necessary tests: clippy, unit tests, and integration tests
+test:
+    sh scripts/check-clippy-version.sh
+    cargo fmt
+    cargo clippy
+    cargo test --workspace
+
+# These tests are a subset of 'cargo test --workspace'
+# It runs ./slosh_test/tests/slosh-tests.rs which loads /slosh_test/run-tests.slosh which loads up all the globals and executes the Example block of each docstring
+# Test docstring examples of slosh globals defined with add_builtin or add_special
+test-slosh-globals:
+     cargo test --package slosh_test --test slosh-tests run_slosh_tests -- --exact --nocapture
+
+# These tests are a subset of 'cargo test --workspace'
+# Test slosh tests defined in /slosh/tests/*.slosh
+test-slosh-tests:
+    cargo test --package slosh --test lisp-scripts -- --nocapture
+
+# Use cspell to spellcheck the codebase
+spellcheck:
+    npx cspell .
```

Comment:
how do i know i have installed everything I need to run just.

do i install just via cargo?
what is npx?

maybe add something in the README so we know how to approach. I like the idea of a centralized file that I can start sticking useful project stuff in, and the shell script to check the cargo version is quite clever!

Reply:
I am fine with this, I don't want to required build deps beyond cargo but I don't think this does that.  Just supports another option plus records some handy command.

Reply:
Once you've [installed](https://github.com/casey/just?tab=readme-ov-file#packages) just, you can prove it by running `just -V` in your shell.

npx is another command-line utility, an alteration of npm the node package manager that comes with nodejs. Running `npx cspell .` downloads [cspell](https://www.npmjs.com/package/cspell) into a local cache on your machine and then runs it. And if it already existed in your local cache then of course it doesn't have to download it. This makes it quite simple to make use of nodejs scripts.

Reply:
It's also possible to [set up shell completion scripts[(https://github.com/casey/just?tab=readme-ov-file#shell-completion-scripts) which makes it quite quick to see the potential commands